### PR TITLE
Add E-Mail Verification, Local Login now works

### PR DESCRIPTION
**Pull request by smith64fx**
Adding stripe support,
Adding E-Mail Verification,
Local Login now works,
added better error handling in jade files

### Description

Fix the bug, that we cannot do a local login because there was no way to do an email verification.

#### Related issues

- #88 
- #32 

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: server/server.js [hunk 1]
```diff
@@ -155,8 +162,12 @@ app.get('/auth/logout', function(req, res, next) {
   res.redirect('/');
 });
 
+app.get('/verified', function(req, res) {
+  res.render('verified');
+});
+
 app.start = function() {
-  // start the web server
+// start the web server
```
Comment (superkhau) [inline server/server.js]: fix indent

File: server/server.js [hunk 2]
```diff
@@ -168,7 +179,22 @@ app.start = function() {
...
+
+  passportConfigurator.setupModels({
+    userModel: app.models.user,
+    userIdentityModel: app.models.userIdentity,
+    userCredentialModel: app.models.userCredential,
+  });
+
+  for (var s in config) {
+    var c = config[s];
+    c.session = c.session !== false;
```
Comment (superkhau) [inline server/server.js]: Can we use longer names than s + c?

**Comment by superkhau** [comment]
Can you fix the linting errors too?

**Comment by superkhau** [comment]
Also squash the commits please.

**Comment by smith64fx** [comment]
Hey guys good morning!
I think the pull request is ready to merge!

✅ Squashed the commits.
✅ Make everything linter friendly

File: server/datasources.json [hunk 3]
```diff
@@ -1,6 +1,24 @@
...
+      "type": "smtp",
+      "host": "mail.schmidigital.de",
+      "secure": false,
+      "port": 587,
+      "tls": {
+        "rejectUnauthorized": false
+      },
+      "auth": {
+        "user": "michael@mindbabble.com",
+        "pass": "25S03&91"
```
Comment (superkhau) [inline server/datasources.json]: Myabe change this to `your-username@your-email.com`? And `pass` to `your-password`?

File: server/datasources.json [hunk 4]
```diff
@@ -1,6 +1,24 @@
...
-    "connector": "memory"
+    "connector": "memory",
+    "file":"db.json"
+  },
+  "emailDs": {
+    "name": "emailDs",
+    "connector": "mail",
+    "transports": [{
+      "type": "smtp",
+      "host": "mail.schmidigital.de",
```
Comment (superkhau) [inline server/datasources.json]: `your-host`

**Comment by superkhau** [comment]
@bajtos Second opinion please.

**Comment by bajtos** [comment]
> Second opinion please. This PR LGTM.

@superkhau I am afraid I am not familiar with this repository. [Commit history](https://github.com/strongloop/loopback-example-passport/graphs/contributors) shows that @raymondfeng, @jannyHou and @loay have the most knowledge - could one of you take a look please?

File: providers.json.template [hunk 5]
```diff
@@ -51,6 +51,19 @@
     "consumerSecret": "{twitter-consumer-secret}",
     "failureFlash": true
   },
+  "stripe-login": {
+    "provider": "stripe",
+    "module": "passport-stripe",
+    "clientID": ""{stripe-client-id}",
+    "clientSecret": ""{stripe-client-secret}",
```
Comment (jannyHou) [inline providers.json.template]: double `"` here, results in json syntax error

File: providers.json.template [hunk 6]
```diff
@@ -51,6 +51,19 @@
     "consumerSecret": "{twitter-consumer-secret}",
     "failureFlash": true
   },
+  "stripe-login": {
+    "provider": "stripe",
+    "module": "passport-stripe",
+    "clientID": ""{stripe-client-id}",
```
Comment (jannyHou) [inline providers.json.template]: double `"` here, results in json syntax error

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (jannyHou) [inline .vscode/settings.json]: debug file accidently uploaded: `.vscode/settings.json`

File: server/views/response.ejs [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<h1><%= title %></h1>
```
Comment (jannyHou) [inline server/views/response.ejs]: nitpick: shall we use `jade` only for consistency? To make ejs work in loopback we might need to setup some config, details see: https://loopback.io/doc/en/lb2/Tutorial-access-control.html#configure-server-side-views

and same for `verify.ejs` and `verified.ejs`

**Pull request closed without merging**